packages/components/src/tabs/index.tsx

```diff
@@ -40,147 +50,41 @@ export const Tabs = Object.assign(
-		const tabsHavePopulatedRef = useRef( false );
-		if ( items.length > 0 ) {
-			tabsHavePopulatedRef.current = true;
-		}
-
-		const selectedTab = items.find( ( item ) => item.id === selectedId );
-		const firstEnabledTab = items.find( ( item ) => {
-			// Ariakit internally refers to disabled tabs as `dimmed`.
-			return ! item.dimmed;
-		} );
-		const initialTab = items.find(
-			( item ) => item.id === `${ instanceId }-${ defaultTabId }`
-		);
-
-		// Handle selecting the initial tab.
-		useLayoutEffect( () => {
-			if ( isControlled ) {
-				return;
-			}
-
-			// Wait for the denoted initial tab to be declared before making a
-			// selection. This ensures that if a tab is declared lazily it can
-			// still receive initial selection, as well as ensuring no tab is
-			// selected if an invalid `defaultTabId` is provided.
-			if ( defaultTabId && ! initialTab ) {
-				return;
-			}
-
-			// If the currently selected tab is missing (i.e. removed from the DOM),
-			// fall back to the initial tab or the first enabled tab if there is
-			// one. Otherwise, no tab should be selected.
-			if ( ! items.find( ( item ) => item.id === selectedId ) ) {
-				if ( initialTab && ! initialTab.dimmed ) {
-					setSelectedId( initialTab?.id );
-					return;
-				}
-
-				if ( firstEnabledTab ) {
-					setSelectedId( firstEnabledTab.id );
-				} else if ( tabsHavePopulatedRef.current ) {
-					setSelectedId( null );
-				}
-			}
-		}, [
-			firstEnabledTab,
-			initialTab,
-			defaultTabId,
-			isControlled,
-			items,
-			selectedId,
-			setSelectedId,
-		] );
-
-		// Handle the currently selected tab becoming disabled.
-		useLayoutEffect( () => {
-			if ( ! selectedTab?.dimmed ) {
-				return;
-			}
-
-			// In controlled mode, we trust that disabling tabs is done
-			// intentionally, and don't select a new tab automatically.
-			if ( isControlled ) {
-				setSelectedId( null );
-				return;
-			}
-
-			// If the currently selected tab becomes disabled, fall back to the
-			// `defaultTabId` if possible. Otherwise select the first
-			// enabled tab (if there is one).
-			if ( initialTab && ! initialTab.dimmed ) {
-				setSelectedId( initialTab.id );
-				return;
-			}
-
-			if ( firstEnabledTab ) {
-				setSelectedId( firstEnabledTab.id );
-			}
-		}, [
-			firstEnabledTab,
-			initialTab,
-			isControlled,
-			selectedTab?.dimmed,
-			setSelectedId,
-		] );
-
-		// Clear `selectedId` if the active tab is removed from the DOM in controlled mode.
-		useLayoutEffect( () => {
-			if ( ! isControlled ) {
-				return;
-			}
-
-			// Once the tabs have populated, if the `selectedTabId` still can't be
-			// found, clear the selection.
-			if (
-				tabsHavePopulatedRef.current &&
-				!! selectedTabId &&
-				! selectedTab
-			) {
-				setSelectedId( null );
-			}
-		}, [ isControlled, selectedTab, selectedTabId, setSelectedId ] );
+		const { items, activeId } = Ariakit.useStoreState( store );
+		const { setActiveId } = store;
 
 		useEffect( () => {
```

Comment:
The behaviour applied by this custom hook is currently being in https://github.com/ariakit/ariakit/issues/4213 , to understand whether it will be even incorporated into ariakit, or if it's even a reasonable fix to implement on our side in the first place.

It will be kept around for the time being.